app/controllers/my-projects/project/hearing/add.js

```diff
@@ -1,4 +1,106 @@
+
+  // to check whether hearing model attributes "location" and "date" are populated
+  // determines certain visual displays based on whether a hearing is submitted
+  hearingSubmitted = false;
+
+  // changes allActions to true or false based on a radio button selection
+  @action
+  setProp(property, newVal) {
+    this.set(property, newVal);
+  }
+
+  @action
+  closeModal() {
+    this.set('modalOpen', false);
+  }
+
+  // checks that the user had input valid values
+  // opens up the modal so the user can review their inputs
+  @action
+  checkInfo() {
+    // set to true for a visual display of which inputs are invalid
+    // passed down to date-time-picker and location-input components
+    this.set('checkIfMissing', true);
+
+    const model = this.get('model');
+
+    const projectDispositions = model.dispositions;
+    const dispositionForAllActions = this.get('dispositionForAllActions');
+
+    let fieldsFilled;
+
+    const allActions = this.get('allActions');
+
+    // a function to check if each hearing location/date field is truthy
+    function infoExists(hearingInfo) {
+      return hearingInfo;
+    }
+
+    if (allActions) {
+      const dispositionLocation = dispositionForAllActions.publichearinglocation;
+      const dispositionDate = dispositionForAllActions.dateofpublichearing;
+      fieldsFilled = dispositionLocation && dispositionDate;
+    } else {
+      const dispositionLocations = projectDispositions.map(disp => `${disp.publichearinglocation}`);
+      const dispositionDates = projectDispositions.map(disp => disp.dateofpublichearing);
+      // using function infoExists, fieldsFilled checks whether each item in array is truthy
+      fieldsFilled = dispositionLocations.every(infoExists) && dispositionDates.every(infoExists);
+    }
+
+    if (fieldsFilled) {
+      this.set('modalOpen', true);
+    }
+  }
+
+  // action triggered when click on the Confirm button in the modal
+  // saves date and location to the disposition model
+  @action
+  async submitHearing() {
+    const model = this.get('model');
+    const allActions = this.get('allActions');
+
+    try {
+      if (allActions) {
+        const projectDispositions = model.dispositions;
+        const allDispositionsLocation = this.get('dispositionForAllActions.publichearinglocation');
+        const allDispositionsDate = this.get('dispositionForAllActions.dateofpublichearing');
+
+        projectDispositions.forEach(function(disposition) {
+          disposition.set('publichearinglocation', allDispositionsLocation);
+          disposition.set('dateofpublichearing', allDispositionsDate);
+        });
+
+        await model.dispositions.save();
+      } else {
+        await model.dispositions.save();
+      }
+      // for displaying certain elements
+      this.set('hearingSubmitted', true);
+      this.set('modalOpen', false);
+    } catch (e) {
+      alert('Sorry about that, unable to connect to server, can you try again?'); // eslint-disable-line
```

Comment:
I'm also averse to using `alert()`, since, like modals, serializes the user's UI.  
In recommendations I assign the error message to a property on the controller. E.g.
```
// ../add.js
this.set('error', 'Sorry about that, unable to connect to server, can you try again?')
```
And then in the template, I render this error if its set
```
// ../add.hbs
{{#if this.error}}
    {{this.error}}
{{/if}}
```
@allthesignals  thoughts?

Reply:
@godfreyyeung that works — you might even just set the error object itself. We don't know if it's a connection issue or an authentication. See other parts of the source code that extract error messages from a `catch`